src/v3/src/components/InputPassword/InputPassword.tsx

```diff
@@ -45,33 +50,61 @@ const InputPassword: UISchemaElementComponent<UISchemaElementComponentWithValida
   const focusRef = useAutoFocus<HTMLInputElement>(focus);
   const hasErrors = typeof errors !== 'undefined';
 
-  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+  const [showPassword, setShowPassword] = useState<boolean>(false);
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
     const changedVal = e.currentTarget.value;
     setTouched?.(true);
     onChangeHandler(changedVal);
     onValidateHandler?.(setErrors, changedVal);
   };
 
+  const handleClickShowPassword = () => {
+    setShowPassword(!showPassword);
+  };
+
+  const handleMouseDownPassword = (e: ClickEvent) => {
+    e.preventDefault();
+  };
```

Comment:
Is this necessary? My initial thought is not really?

Reply:
This was also transferred from the odyssey component, and prevents the input field from losing focus when clicking the button.

Reply:
Ohh, this is tied to the toggle visibility button, ok perfect! Ty!